plugins/activity/plugin.go

```diff
@@ -48,31 +55,49 @@ func (svc *Plugin) Init(temporal temporal.Temporal, server server.Server, log lo
 func (svc *Plugin) Serve() chan error {
 	errCh := make(chan error, 1)
 
-	pool, err := NewActivityPool(context.Background(), *svc.temporal.GetConfig().Activities, svc.server)
+	pool, err := svc.startPool()
 	if err != nil {
-		errCh <- errors.E(errors.Op("newActivityPool"), err)
-		return errCh
-	}
-
-	// todo: proxy events
-
-	err = pool.Start(context.Background(), svc.temporal)
-	if err != nil {
-		errCh <- errors.E(errors.Op("startActivityPool"), err)
+		errCh <- errors.E("startPool", err)
 		return errCh
 	}
 
 	svc.pool = pool
 
-	svc.log.Debug("Started activity processing", zap.Any("activities", pool.activities))
+	go func() {
+		for {
+			select {
+			case <-svc.reset:
+				if atomic.LoadInt64(&svc.closing) == 1 {
+					return
+				}
+
+				err := svc.replacePool()
+				if err == nil {
+					continue
+				}
+
+				bkoff := backoff.NewExponentialBackOff()
+
+				err = backoff.Retry(svc.replacePool, bkoff)
+				if err != nil {
+					errCh <- errors.E("deadPool", err)
```

Comment:
`const`, allocate once before for loop
Also, for a single case `select`, to simplify, use `for _ := range svc.reset`